core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1667,6 +1673,13 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       if (key != null) {
         addedJars(key) = System.currentTimeMillis
         logInfo("Added JAR " + path + " at " + key + " with timestamp " + addedJars(key))
+        if (addToCurrentThread) {
+          val currentCL = Utils.getContextOrSparkClassLoader
+          currentCL match {
+            case cl: MutableURLClassLoader => cl.addURL(new URI(key).toURL())
+            case _ => logWarning(s"Unsupported cl $currentCL will not update jars thread cl")
```

Comment:
My opinion is that it's better to throw an exception here. Though as that's an API behavior question, probably best to defer to a core maintainer? @pwendell any thoughts? And thoughts on adding this option to addJar in general?